### PR TITLE
Some changes

### PR DESCRIPTION
**Pull request by MichaelDvP**
I've split in some commits, or should i make different PRs?
I'm actually on ESP32 and have tested all. 
I rechecked  json sizes for boiler and thermostat and try to use a single json on the stack and clear it after first output of the spilt messages (main/ww, main/hc).
Please check what is usefull for the comming release and what should go to next dev.

### CODE REVIEW

**Review by proddy — APPROVED** [review]
nice additions.

**Review by proddy — APPROVED** [review]
only comment is to mention the 0xFF in the troubleshooting guide

File: src/devices/boiler.cpp [hunk 1]
```diff
@@ -1022,10 +1039,29 @@ void Boiler::process_UBAMaintenanceStatus(std::shared_ptr<const Telegram> telegr
     // first byte: Maintenance due (0 = no, 3 = yes, due to operating hours, 8 = yes, due to date)
 }
 
-// 0x10, 0x11, 0x12
+// 0x10, 0x11
 // not yet implemented
```
Comment (proddy) [inline src/devices/boiler.cpp:1043]: line 1043 can go since you implemented it!

File: src/devices/switch.cpp [hunk 2]
```diff
@@ -28,28 +28,134 @@ uuid::log::Logger Switch::logger_ {
...
+    JsonObject dev = doc.createNestedObject("dev");
+    dev["name"]    = F("EMS-ESP Switch");
+    dev["sw"]      = EMSESP_APP_VERSION;
+    dev["mf"]      = this->brand_to_string();
+    dev["mdl"]     = this->name();
+    JsonArray ids  = dev.createNestedArray("ids");
+    ids.add("ems-esp-switch");
+
+    Mqtt::publish_retain(F("homeassistant/sensor/ems-esp/switch/config"), doc.as<JsonObject>(), true); // publish the config payload with retain flag
+    Mqtt::register_mqtt_ha_sensor(PSTR("switch"), nullptr, F_(flowTemp), this->device_type(), "activated", nullptr, nullptr);
```
Comment (proddy) [inline src/devices/switch.cpp:143]: add to locale_en and use F_(...) ?

File: src/devices/switch.cpp [hunk 3]
```diff
@@ -28,28 +28,134 @@ uuid::log::Logger Switch::logger_ {
...
+    Mqtt::register_mqtt_ha_sensor(PSTR("switch"), nullptr, F_(flowTemp), this->device_type(), "flowTemp", F_(degrees), F_(icontemperature));
+    Mqtt::register_mqtt_ha_sensor(PSTR("switch"), nullptr, F_(flowTemp), this->device_type(), "status", nullptr, nullptr);
+
+    mqtt_ha_config_ = true; // done
+}
+
+// message 0x9B switch on/off
```
Comment (proddy) [inline src/devices/switch.cpp:150]: can you provide example telegams for 09B and 0x9C so I can add to test.cpp. Or you can do it yourself. Then I can simulate the behaviour in the web, HA without actually having a device.

File: src/emsesp.cpp [hunk 4]
```diff
@@ -644,6 +648,9 @@ bool EMSESP::process_telegram(std::shared_ptr<const Telegram> telegram) {
 
     if (!found) {
         LOG_DEBUG(F("No telegram type handler found for ID 0x%02X (src 0x%02X)"), telegram->type_id, telegram->src);
+        if ((watch() == WATCH_ON) && (watch_id_ == 0xFF)) {
```
Comment (proddy) [inline src/emsesp.cpp:651]: prefer if this was a WATCH_UNKNOWN instead of hard-coded

File: src/devices/thermostat.h [hunk 5]
```diff
@@ -170,6 +170,7 @@ class Thermostat : public EMSdevice {
     uint8_t ibaClockOffset_       = EMS_VALUE_UINT_NOTSET; // offset (in sec) to clock, 0xff = -1 s, 0x02 = 2 s
 
     uint16_t errorNumber_        = EMS_VALUE_USHORT_NOTSET;
+    char     lastCode_[30]       = {'\0'};
```
Comment (proddy) [inline src/devices/thermostat.h:173]: isn't last code only for the Boiler, not the thermostat?

Comment (MichaelDvP) [reply src/devices/thermostat.h:173]: Boiler stores in message 0x10 and 0x11, for my boiler the last is in 0x10, 0x11 is older, but #614 reports last in 0x11. 
Thermostat has it's own storage in message 0x12, mainly communication errors.
In my system (output from this code):
Boiler: `  Last error: 6L(548) 24.01.2020 02:52`
Thermostat: ` Last error: 22(816) 17.11.2020 10:44`

File: src/devices/switch.cpp (hunk 3 above)
Comment (MichaelDvP) [reply src/devices/switch.cpp:150]: Sorry, i don't have WM10, it's from [https://emswiki.thefischer.net/doku.php?id=wiki:ems:telegramme#wm10parameter](https://emswiki.thefischer.net/doku.php?id=wiki:ems:telegramme#wm10parameter) as i searched for other telegrams, but is see that WM10 is already a recognized device.

File: src/devices/switch.cpp (hunk 2 above)
Comment (MichaelDvP) [reply src/devices/switch.cpp:143]: Ups, these register_mqtt_ha_sensor have some more copy&past errors.

Comment (MichaelDvP) [reply src/devices/switch.cpp]: should be fixed now.

**Pull request merged**